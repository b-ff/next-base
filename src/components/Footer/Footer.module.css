.footer {
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid var(--color_border_primary);
  padding: var(--spacing_4x) var(--spacing_4x);
  color: var(--color_font_neutral);
  font-size: var(--font_size_small);
}

.footer a {
  color: var(--color_font_neutral);
  text-decoration: none;
  border-bottom: 1px dashed var(--color_font_neutral);
}
