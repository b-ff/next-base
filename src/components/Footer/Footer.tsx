"use client";
import { FC, ReactNode } from "react";
import styles from "./Footer.module.css";

export type FooterProps = {
  children?: ReactNode;
};

/**
 * Footer of the app
 */
export const Footer: FC<FooterProps> = ({ children }) => {
  return (
    <header className={styles.footer}>
      {children}
      <span>
        <a
          href="https://github.com/b-ff"
          target="_blank"
          rel="noopener noreffer"
        >
          Slava Biryukov
        </a>{" "}
        x {new Date().getFullYear()} {children}
      </span>
    </header>
  );
};
