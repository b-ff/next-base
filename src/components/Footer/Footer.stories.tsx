import type { Meta, StoryObj } from "@storybook/react";

import { Footer } from "./Footer";

const meta: Meta<typeof Footer> = {
  component: Footer,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Footer>;

export const Primary: Story = {
  args: {
    children: "Content",
  },
};
