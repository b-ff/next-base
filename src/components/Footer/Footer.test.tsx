import { Footer } from "./Footer";
import { render } from "@testing-library/react";

const mockChildren = "Content";

describe("Component Footer", () => {
  it("Should render as expected", () => {
    const { container } = render(<Footer>{mockChildren}</Footer>);
    expect(container).toMatchSnapshot();
  });
});
