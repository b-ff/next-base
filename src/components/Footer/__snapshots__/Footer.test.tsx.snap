// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Footer Should render as expected 1`] = `
<div>
  <header
    class="footer"
  >
    Content
    <span>
      <a
        href="https://github.com/b-ff"
        rel="noopener noreffer"
        target="_blank"
      >
        Slava Biryukov
      </a>
       
      x 
      2023
       
      Content
    </span>
  </header>
</div>
`;
